### PR TITLE
DTPPCPSDK-2227: Return V2 Variables from getHostedButtonDetails 👾

### PR DESCRIPTION
**Pull request by spencersablan**
### Description

- 🥖 **Prepares us for the NCPS 2.0 changes** - All we need to do for this PR is extract those variables from the response to prepare for future changes.
- 📝  **Added tests for v2 response** -  Since we need to support both V1 & V2 responses from `/ncp/api/form-fields/`, I added some quick unit tests to make sure `getHostedButtonDetails` can handle either response appropriately.

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
**JIRA:** [DTPPCPSDK-2227](https://paypal.atlassian.net/browse/DTPPCPSDK-2227)

### Groups who should review (if applicable)

❤️ Thank you!

### CODE REVIEW

**Review by jshawl — APPROVED** [review]
:shipit:

File: src/hosted-buttons/types.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ export type HostedButtonsInstance = {|
   render: (string | HTMLElement) => Promise<void>,
 |};
 
+export type HostedButtonPreferences = {|
+  buttonPreferences: $ReadOnlyArray<string>,
```
Comment (imbrian) [inline src/hosted-buttons/types.js:38]: Possible to narrow these `string` types to known preferences and sources? Could do that in follow-up implementation PR too

**Review by imbrian — APPROVED** [review]
small comment re: preference types

File: src/hosted-buttons/types.js (hunk 1 above)
Comment (spencersablan) [reply src/hosted-buttons/types.js:38]: I like that!

Comment (spencersablan) [reply src/hosted-buttons/types.js]: Donezo funzo

**Review by nbierdeman — APPROVED** [review]
Nice! 🤘 🚀 Should we add `tagline` to the tests?

**Comment by imbrian** [comment]
> Nice! 🤘 🚀 Should we add `tagline` to the tests?

`tagline` isn't implemented; fix is [here](https://github.com/paypal/paypal-checkout-components/pull/2384), but waiting for @spencersablan to merge this to deal with conflicts on that branch.

**Pull request merged**